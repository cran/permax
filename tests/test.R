# attach the permax library before sourcing this file
#library(permax,lib.loc='/usr/stats/gray/chips')
#library(permax,lib.loc='~/R')
library(permax)
#dyn.load('../src/permax.so')
#source('../R/permax.R')
#generate make believe data
set.seed(1292)
ngenes <- 1000
m1 <- rnorm(ngenes,4,1)
m2 <- rnorm(ngenes,4,1)
exp1 <- cbind(matrix(exp(rnorm(ngenes*5,m1,1)),nrow=ngenes),
               matrix(exp(rnorm(ngenes*10,m2,1)),nrow=ngenes))
exp1[exp1<20] <- 20
sub <- exp1>20 & exp1<150
exp1[sub] <- ifelse(runif(length(sub[sub]))<.5,20,exp1[sub])
dimnames(exp1) <- list(paste('x',1:ngenes,sep=''),
                       paste('sample',1:ncol(exp1),sep=''))
exp1 <- round(exp1)
print(table(exp1<=20))
n1 <- apply(exp1>20,1,sum)
print(table(n1))
# permax
uu <- permax(exp1,1:5)
print(uus1 <- summary(uu))
# following should give same results
exp2 <- exp1[,c(1,2,7,10,3,8,11,6,5,9,12,15,13,14,4)]
exp3 <- cbind(ngenes:1,exp1,1:ngenes)
exp4 <- cbind(ngenes:1,exp2,1:ngenes)
uu2 <- permax(exp2,c(1,2,5,9,15))
uus2 <- summary(uu2)
uu3 <- permax(exp3,2:6,ig2=7:16)
uus3 <- summary(uu3)
uu4 <- permax(exp4,c(1,2,5,9,15)+1,ig2=c(4,5,7,8,9,11,12,13,14,15))
uus4 <- summary(uu4)
range(as.matrix(uus2$objectl)-as.matrix(uus1$objectl))
range(as.matrix(uus3$objectl)-as.matrix(uus1$objectl))
range(as.matrix(uus4$objectl)-as.matrix(uus1$objectl))
range(as.matrix(uus2$objectr)-as.matrix(uus1$objectr))
range(as.matrix(uus3$objectr)-as.matrix(uus1$objectr))
range(as.matrix(uus4$objectr)-as.matrix(uus1$objectr))

postscript(file='figs.ps')
par(cex=.7,mar=c(6,6,2,2))
plot(uu,exp1,ig1=1:5)
plot(uu,exp1,ig1=1:5,nr=40,nl=0)
plot.expr(exp1[1:20,])
dev.off()

u4 <- permax(exp1,1:5, nperm=1000)
print(summary(u4))
print(attr(u4,'seed.start'))
print(attr(u4,'seed.end'))

#clustered
clustind <- c(1,1,2,2,3,1,1,1,1,2,2,2,2,3,3)
uuc <- permax(exp1,1:5,nperm=5000,cluster=clustind,WHseed=attr(u4,'seed.end'))
summary(uuc,nl=5,nr=5) 
# stratified on cluster, using ranks; equal weights
uus <- permax(exp1,1:5,nperm=5000,cluster=clustind,stratify=TRUE,
          ranks=TRUE,WHseed=attr(uuc,'seed.end'))
summary(uus,nl=5,nr=5) 
print(attr(uus,'seed.end'))
# stratified on cluster, using ranks; average weighted by cluster sizes
uus <- permax(exp1,1:5,nperm=5000,cluster=clustind,stratify=TRUE,
           ranks=TRUE,weights=table(clustind),WHseed=attr(uus,'seed.end'))
summary(uus,nl=5,nr=5) 
print(attr(uus,'seed.start'))
print(attr(uus,'seed.end'))
summary(uus,nl=10,nr=0) 

clust2 <- c(1,1,2,2,2,3,3,4,4,5,5,5,6,6,6)
uu5 <- permax(exp1,1:5,nperm=5000,cluster=clust2,permute.cluster=TRUE,
           ranks=TRUE,WHseed=attr(uus,'seed.end'))
summary(uu5,nl=5,nr=5)
uu6 <- permax(exp1,1:5,nperm=0,cluster=clust2,permute.cluster=TRUE,ranks=TRUE)
print(uu6s <- summary(uu6,nl=5,nr=5))
apply(exp1[uu6s$upper,],1,rank)
uu6[uu6s$upper,]$pind.upper

ucor <- permcor(exp1[,3:8],1:6,nperm=0)
print(summary(ucor))
ucor <- permcor(exp1,1:15,WHseed=attr(uus,'seed.end'))
print(summary(ucor))
print(attr(ucor,'seed.end'))

ucorx <- permcor(exp1[,4:7],1:4,nperm=0)
summary(ucorx,nl=5,nr=5)
ucorx <- permcor(exp1[,4:7],1:4,nperm=0,cluster=c(1,2,1,2))
summary(ucorx,nl=5,nr=5)
ucorx <- permcor(exp1[,3:9],1:7,nperm=0,cluster=c(1,2,3,1,2,3,3))
summary(ucorx,nl=5,nr=5)
ucorx <- permcor(exp1[,3:9],1:7,nperm=10,cluster=c(1,2,3,1,2,3,3),WHseed=attr(uus,'seed.end'))
summary(ucorx,nl=5,nr=5)
ucorx <- permcor(exp1[,3:10],c(1,2,2,3,3,4,4,4),nperm=10,cluster=c(1,2,2,3,3,4,4,4),permute.cluster=TRUE,WHseed=attr(uus,'seed.end'))
summary(ucorx,nl=5,nr=5)
ucorx <- permcor(exp1[,3:10],c(1,2,2,3,3,4,4,4),nperm=0,cluster=c(1,2,2,3,3,4,4,4),permute.cluster=TRUE)
summary(ucorx,nl=5,nr=5)


postscript(file='figs2.ps')
par(cex=.7,mar=c(6,6,2,2))
plot(ucor,exp1) #columns of exp1 already sorted on Z
dev.off()

u8 <- permcor(exp1,1:15,cluster=clustind,WHseed=attr(ucor,'seed.end'))
summary(u8,nr=4,nl=4)
# correlations estimated within clusters; average weighted by cluster sizes
u8 <- permcor(exp1,1:15,cluster=clustind,stratify=TRUE,
              weights=table(clustind),WHseed=attr(u8,'seed.end'))
summary(u8,nr=4,nl=4)

# should give same p-values:
u1 <- permax(exp1[,c(1:3,5:8)],1:3)
u2 <- permcor(exp1[,c(1:3,5:8)],c(1,1,1,0,0,0,0),nperm=0)
print(range(u1$p.lower-u2$p.lower))
print(range(u1$p.upper-u2$p.upper))

uuu <- permsep(exp1,1:5)
print(uuu$dtcs)
uuu <- permsep(exp2,c(1,2,5,9,15))
uuu <- permsep(exp3,2:6,ig2=7:16)
uuu <- permsep(exp4,c(1,2,5,9,15)+1,ig2=c(4,5,7,8,9,11,12,13,14,15))
uuu <- permsep(exp1,1:5,nperm=1000)
print(attr(uuu,'seed.start'))
print(attr(uuu,'seed.end'))
u5 <- permsep(exp1,ig1=1:5,nperm=1000,WHseed=attr(uuu,'seed.end'))
print(attr(u5,'seed.start'))
print(attr(u5,'seed.end'))
uuu <- permsep(exp1,1:5,cluster=clustind,nperm=10000,WHseed=attr(u5,'seed.end'))
uuu <- permsep(exp1,1:5,nperm=2000,cluster=clustind,stratify=TRUE,
          WHseed=attr(uuu,'seed.end'))
uuu <- permsep(exp1,1:5,cluster=clust2,nperm=1000,WHseed=attr(uuu,'seed.end'),permute.cluster=TRUE)
uuu <- permsep(exp1,1:5,cluster=clust2,nperm=0,permute.cluster=TRUE)

x <- matrix(c(1:10,NA,NA),4,dimnames=list(format(1:4),c('a','b','c')))
impmv(x)
na.omit(impmv(x,min.nonmiss=3))
irn <- function(x,nc) rnorm(nc-length(x),mean(x),sqrt(var(x)))
impmv(x,IFUN=irn,nc=ncol(x)) 
impmv(t(x),IFUN=irn,nc=nrow(x)) 

x <- matrix(1:12,3)
rowperm(x)

# paired data -- checked the p-values and stats by hand
clust <- c(1,2,3,4,1,2,3,4)
dat <- rbind(c(1:8),c(8,1:7),c(5,1:7),c(8,7,1:6))
dimnames(dat) <- list(1:4,1:8)
permax(dat,1:4,cluster=clust,signed.rank=TRUE)
# note, order of the rows is flipped in the output of the following call
permax(dat,5:8,cluster=clust,signed.rank=TRUE)
