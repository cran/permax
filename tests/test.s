# attach the permax library before sourcing this file
#library(permax,lib.loc='/usr/stats/gray/chips')

#generate make believe data
set.seed(1292)
ngenes <- 1000
m1 <- rnorm(ngenes,4,1)
m2 <- rnorm(ngenes,4,1)
exp1 <- cbind(matrix(exp(rnorm(ngenes*5,m1,1)),nrow=ngenes),
               matrix(exp(rnorm(ngenes*10,m2,1)),nrow=ngenes))
exp1[exp1<20] <- 20
sub <- exp1>20 & exp1<150
exp1[sub] <- ifelse(runif(length(sub[sub]))<.5,20,exp1[sub])
dimnames(exp1) <- list(paste('x',1:ngenes,sep=''),
                       paste('sample',1:ncol(exp1),sep=''))
exp1 <- round(exp1)
print(table(exp1<=20))
n1 <- apply(exp1>20,1,sum)
print(table(n1))
# permax
uu <- permax(exp1,1:5)
print(summary(uu))
#plots
ps.options(image.colors=cbind(seq(0,1,length=30),seq(0,1,length=30),seq(1,0,length=30)))
postscript(file='figs.ps')
#motif("-xrm 'sgraphMotif.colorSchemes : background : black; lines : yellow cyan magenta green MediumBlue red; text : white yellow cyan magenta green MediumBlue red; images : blue 30 yellow'")
#motif("-xrm 'sgraphMotif.defaultColorScheme : 3'") # cyan to red
#graphsheet(file='f:\\chips\\fig1.jpg',format='jpg',num.image.colors=2,
#  num.image.shades=30,image.color.table="0,0,255|255,255,0")

plot(uu,exp1,ig1=1:5,cex=.7)
plot.expr(exp1[1:20,])
dev.off()

u4 <- permax(exp1,1:5, nperm=1000)
print(summary(u4))
print(attr(u4,'seed.start'))
print(attr(u4,'seed.end'))

u5 <- permsep(exp1,ig1=1:5,nperm=1000,WHseed=attr(u4,'seed.end'))
print(attr(u5,'seed.start'))
print(attr(u5,'seed.end'))

ucor <- permcor(exp1[,3:8],1:6,nperm=0)
print(summary(ucor))
ucor <- permcor(exp1,1:15)
print(summary(ucor))
print(attr(ucor,'seed.end'))
#ucor <- permcor(exp1[1:50,1:5],1:5,nperm=500)
#for (i in 1:5) print(cor(log(exp1[i,1:5]),1:5))

# should give same p-values:
u1 <- permax(exp1[,c(1:3,5:8)],1:3)
u2 <- permcor(exp1[,c(1:3,5:8)],c(1,1,1,0,0,0,0),nperm=0)
print(range(u1$p.lower-u2$p.lower))
print(range(u1$p.upper-u2$p.upper))

uuu <- permsep(exp1,1:5)
print(uuu$dtcs)
uuu <- permsep(exp1,1:5,nperm=1000)
