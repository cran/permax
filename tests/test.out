 FALSE TRUE 
  7490 7510
 0 1  2  3  4  5   6   7   8   9 10 11 12 13 14 15 
 3 9 34 48 77 97 116 120 123 115 83 78 50 28 16  3
statistics will be computed for all 3003 groupings
       stat  p.lower   m1   m2     s1    s2 np1 np2 mdiff   mrat 
x796 -13.81 0.000333 3.26 6.74 0.3594 0.499   2  10  -819 0.0307
 x48  -8.83 0.001665 3.07 6.13 0.1662 0.753   1  10  -438 0.0468
x845  -7.53 0.005994 3.00 6.16 0.0000 0.922   0  10  -453 0.0423
x194  -7.27 0.007326 3.73 7.26 0.7737 0.935   3  10 -1387 0.0291
x165  -7.25 0.007326 3.36 6.55 0.4831 0.912   3  10  -672 0.0410
x978  -6.29 0.014652 3.25 5.95 0.5729 0.861   1  10  -359 0.0672
x499  -6.23 0.015318 3.00 6.17 0.0000 1.119   0  10  -458 0.0418
x151  -5.92 0.022311 3.38 6.43 0.8540 0.977   1  10  -590 0.0473
x268  -5.88 0.023310 3.69 6.44 0.8341 0.861   3  10  -585 0.0641
x432  -5.81 0.025974 3.50 6.39 1.0117 0.857   2  10  -562 0.0557
x484  -5.64 0.030303 3.59 5.89 0.8218 0.706   2  10  -324 0.1005
x860  -5.53 0.033966 3.68 5.53 0.4434 0.669   4  10  -212 0.1583
x240  -5.53 0.033966 3.59 6.34 1.0064 0.857   2  10  -529 0.0644
x233  -5.49 0.035298 3.00 5.30 0.0000 0.920   0   9  -180 0.0998
x619  -5.37 0.043956 3.00 6.04 0.0000 1.242   0   9  -399 0.0477
x767  -5.30 0.047286 3.00 5.46 0.0000 1.019   0   9  -215 0.0852
x135  -5.25 0.048951 3.00 5.35 0.0000 0.984   0  10  -191 0.0948
x502  -5.14 0.056277 3.16 5.90 0.2365 1.156   2   9  -340 0.0650
x189  -4.97 0.066600 3.22 6.47 0.4060 1.410   2   9  -622 0.0387
x389  -4.90 0.073926 3.16 5.47 0.3725 1.003   1  10  -213 0.0998
x391  -4.85 0.076590 3.91 6.05 1.2604 0.477   2  10  -373 0.1183
x567  -4.82 0.080253 3.23 6.32 0.3489 1.383   3   9  -529 0.0458
x429  -4.79 0.082584 3.03 5.32 0.0815 1.044   1   9  -183 0.1020
x270  -4.51 0.111888 3.47 5.58 0.6560 0.927   3  10  -233 0.1218
x368  -4.40 0.129870 3.95 6.16 1.3131 0.669   2  10  -423 0.1095
      stat  p.upper   m1   m2    s1     s2 np1 np2 mdiff  mrat 
x193 24.89 0.000333 5.38 3.04 0.274 0.0963   5   2   196 10.43
x506 24.79 0.000333 5.81 3.01 0.361 0.0577   5   1   313 16.37
x475 13.88 0.002664 5.41 3.11 0.304 0.3013   5   2   200  9.94
 x35 11.58 0.006327 5.78 3.13 0.618 0.2872   5   2   301 14.14
 x66 10.80 0.010989 5.91 3.00 0.887 0.0154   5   1   350 18.40
x631 10.43 0.012987 6.74 3.21 0.848 0.4829   5   2   820 34.22
x643 10.28 0.013653 5.82 3.24 0.489 0.4443   5   3   313 13.23
x243  9.98 0.013653 5.89 3.35 0.264 0.5298   5   6   334 12.66
x255  9.91 0.013986 5.51 3.29 0.335 0.4373   5   4   220  9.18
x304  9.64 0.016317 5.62 3.05 0.841 0.1678   5   1   255 13.10
x183  9.24 0.019314 6.20 3.30 0.676 0.5204   5   3   466 18.13
x339  9.09 0.020646 5.72 3.08 0.916 0.1834   5   2   284 14.05
x130  9.06 0.020646 6.17 3.12 1.052 0.2360   5   3   457 21.20
x595  8.49 0.029970 6.76 3.50 0.683 0.7098   5   4   828 26.07
x545  7.84 0.043290 5.10 3.03 0.861 0.0949   5   1   144  8.00
x844  7.55 0.052947 6.81 3.06 1.618 0.1426   5   2   881 42.21
x498  7.16 0.065934 5.00 3.15 0.393 0.5026   5   1   126  6.35
x402  7.14 0.067599 5.03 3.01 0.926 0.0577   5   1   133  7.51
x358  6.82 0.079920 6.55 3.57 0.754 0.8152   5   5   661 19.65
 x96  6.75 0.084582 5.00 3.01 0.965 0.0577   5   1   128  7.28
x921  6.69 0.088578 5.51 3.14 0.956 0.4447   5   2   224 10.68
x635  6.40 0.104895 7.31 3.72 1.283 0.8875   5   5  1461 36.51
x253  6.33 0.110223 7.77 3.74 1.327 1.0794   5   6  2323 56.14
x250  6.30 0.112554 6.76 3.67 0.627 0.9912   5   4   821 21.91
x405  6.28 0.113553 5.47 3.09 1.192 0.2538   5   3   217 10.89
$lower:
 [1] "x796" "x48"  "x845" "x194" "x165" "x978" "x499" "x151" "x268" "x432"
[11] "x484" "x860" "x240" "x233" "x619" "x767" "x135" "x502" "x189" "x389"
[21] "x391" "x567" "x429" "x270" "x368"

$upper:
 [1] "x193" "x506" "x475" "x35"  "x66"  "x631" "x643" "x243" "x255" "x304"
[11] "x183" "x339" "x130" "x595" "x545" "x844" "x498" "x402" "x358" "x96" 
[21] "x921" "x635" "x253" "x250" "x405"

       stat p.lower   m1   m2     s1    s2 np1 np2 mdiff   mrat 
x796 -13.81   0.000 3.26 6.74 0.3594 0.499   2  10  -819 0.0307
 x48  -8.83   0.001 3.07 6.13 0.1662 0.753   1  10  -438 0.0468
x845  -7.53   0.006 3.00 6.16 0.0000 0.922   0  10  -453 0.0423
x194  -7.27   0.006 3.73 7.26 0.7737 0.935   3  10 -1387 0.0291
x165  -7.25   0.006 3.36 6.55 0.4831 0.912   3  10  -672 0.0410
x978  -6.29   0.015 3.25 5.95 0.5729 0.861   1  10  -359 0.0672
x499  -6.23   0.017 3.00 6.17 0.0000 1.119   0  10  -458 0.0418
x151  -5.92   0.024 3.38 6.43 0.8540 0.977   1  10  -590 0.0473
x268  -5.88   0.024 3.69 6.44 0.8341 0.861   3  10  -585 0.0641
x432  -5.81   0.026 3.50 6.39 1.0117 0.857   2  10  -562 0.0557
x484  -5.64   0.029 3.59 5.89 0.8218 0.706   2  10  -324 0.1005
x860  -5.53   0.032 3.68 5.53 0.4434 0.669   4  10  -212 0.1583
x240  -5.53   0.032 3.59 6.34 1.0064 0.857   2  10  -529 0.0644
x233  -5.49   0.033 3.00 5.30 0.0000 0.920   0   9  -180 0.0998
x619  -5.37   0.041 3.00 6.04 0.0000 1.242   0   9  -399 0.0477
x767  -5.30   0.045 3.00 5.46 0.0000 1.019   0   9  -215 0.0852
x135  -5.25   0.045 3.00 5.35 0.0000 0.984   0  10  -191 0.0948
x502  -5.14   0.048 3.16 5.90 0.2365 1.156   2   9  -340 0.0650
x189  -4.97   0.058 3.22 6.47 0.4060 1.410   2   9  -622 0.0387
x389  -4.90   0.067 3.16 5.47 0.3725 1.003   1  10  -213 0.0998
x391  -4.85   0.071 3.91 6.05 1.2604 0.477   2  10  -373 0.1183
x567  -4.82   0.074 3.23 6.32 0.3489 1.383   3   9  -529 0.0458
x429  -4.79   0.079 3.03 5.32 0.0815 1.044   1   9  -183 0.1020
x270  -4.51   0.102 3.47 5.58 0.6560 0.927   3  10  -233 0.1218
x368  -4.40   0.119 3.95 6.16 1.3131 0.669   2  10  -423 0.1095
      stat p.upper   m1   m2    s1     s2 np1 np2 mdiff  mrat 
x193 24.89   0.000 5.38 3.04 0.274 0.0963   5   2   196 10.43
x506 24.79   0.000 5.81 3.01 0.361 0.0577   5   1   313 16.37
x475 13.88   0.004 5.41 3.11 0.304 0.3013   5   2   200  9.94
 x35 11.58   0.007 5.78 3.13 0.618 0.2872   5   2   301 14.14
 x66 10.80   0.013 5.91 3.00 0.887 0.0154   5   1   350 18.40
x631 10.43   0.015 6.74 3.21 0.848 0.4829   5   2   820 34.22
x643 10.28   0.015 5.82 3.24 0.489 0.4443   5   3   313 13.23
x243  9.98   0.015 5.89 3.35 0.264 0.5298   5   6   334 12.66
x255  9.91   0.017 5.51 3.29 0.335 0.4373   5   4   220  9.18
x304  9.64   0.021 5.62 3.05 0.841 0.1678   5   1   255 13.10
x183  9.24   0.026 6.20 3.30 0.676 0.5204   5   3   466 18.13
x339  9.09   0.027 5.72 3.08 0.916 0.1834   5   2   284 14.05
x130  9.06   0.027 6.17 3.12 1.052 0.2360   5   3   457 21.20
x595  8.49   0.039 6.76 3.50 0.683 0.7098   5   4   828 26.07
x545  7.84   0.058 5.10 3.03 0.861 0.0949   5   1   144  8.00
x844  7.55   0.065 6.81 3.06 1.618 0.1426   5   2   881 42.21
x498  7.16   0.080 5.00 3.15 0.393 0.5026   5   1   126  6.35
x402  7.14   0.082 5.03 3.01 0.926 0.0577   5   1   133  7.51
x358  6.82   0.095 6.55 3.57 0.754 0.8152   5   5   661 19.65
 x96  6.75   0.097 5.00 3.01 0.965 0.0577   5   1   128  7.28
x921  6.69   0.101 5.51 3.14 0.956 0.4447   5   2   224 10.68
x635  6.40   0.122 7.31 3.72 1.283 0.8875   5   5  1461 36.51
x253  6.33   0.123 7.77 3.74 1.327 1.0794   5   6  2323 56.14
x250  6.30   0.126 6.76 3.67 0.627 0.9912   5   4   821 21.91
x405  6.28   0.126 5.47 3.09 1.192 0.2538   5   3   217 10.89
$lower:
 [1] "x796" "x48"  "x845" "x194" "x165" "x978" "x499" "x151" "x268" "x432"
[11] "x484" "x860" "x240" "x233" "x619" "x767" "x135" "x502" "x189" "x389"
[21] "x391" "x567" "x429" "x270" "x368"

$upper:
 [1] "x193" "x506" "x475" "x35"  "x66"  "x631" "x643" "x243" "x255" "x304"
[11] "x183" "x339" "x130" "x595" "x545" "x844" "x498" "x402" "x358" "x96" 
[21] "x921" "x635" "x253" "x250" "x405"

[1] 17264  1341 17250
[1] 14865 24966  5969
# comp sep; prop perm with more; ave # per perm; prop perm with any
[1] 38.000  0.002  0.403  0.245
[1] 14865 24966  5969
[1] 25218  6537  1350
statistics will be computed for all 720 permutationss
       stat p.lower np 
x255 -0.961   0.328  5
 x58 -0.941   0.579  4
x612 -0.934   0.703  4
x161 -0.932   0.721  5
x447 -0.926   0.797  4
x349 -0.926   0.801  4
x844 -0.924   0.814  4
x921 -0.924   0.832  3
x923 -0.923   0.835  3
x250 -0.917   0.890  3
x545 -0.916   0.897  3
x422 -0.913   0.917  4
x193 -0.912   0.918  5
 x35 -0.910   0.933  3
x643 -0.904   0.965  3
x100 -0.903   0.968  4
x554 -0.902   0.976  5
 x11 -0.897   0.983  3
x573 -0.896   0.983  3
x317 -0.892   0.983  3
x339 -0.886   0.983  3
x655 -0.885   0.983  5
x294 -0.883   0.985  4
x630 -0.881   0.986  4
x358 -0.881   0.986  4
      stat p.upper np 
x860 0.967   0.228  5
x281 0.941   0.581  3
x206 0.935   0.675  5
x567 0.934   0.706  4
x732 0.932   0.724  4
x251 0.929   0.781  3
x550 0.926   0.800  3
x915 0.923   0.835  3
x175 0.920   0.874  5
x706 0.918   0.889  3
x246 0.913   0.912  3
x142 0.913   0.918  5
x385 0.911   0.931  3
x331 0.902   0.979  4
x796 0.899   0.982  4
x499 0.898   0.982  3
x437 0.895   0.983  3
x165 0.891   0.983  5
x575 0.890   0.983  3
x541 0.885   0.985  3
x747 0.884   0.985  4
x270 0.884   0.985  5
x938 0.883   0.985  4
x978 0.875   0.988  4
x947 0.874   0.990  4
$lower:
 [1] "x255" "x58"  "x612" "x161" "x447" "x349" "x844" "x921" "x923" "x250"
[11] "x545" "x422" "x193" "x35"  "x643" "x100" "x554" "x11"  "x573" "x317"
[21] "x339" "x655" "x294" "x630" "x358"

$upper:
 [1] "x860" "x281" "x206" "x567" "x732" "x251" "x550" "x915" "x175" "x706"
[11] "x246" "x142" "x385" "x331" "x796" "x499" "x437" "x165" "x575" "x541"
[21] "x747" "x270" "x938" "x978" "x947"

       stat p.lower np 
x255 -0.888   0.008  9
x193 -0.817   0.079  7
x243 -0.816   0.080 11
x844 -0.807   0.106  7
x339 -0.795   0.145  7
x183 -0.794   0.150  8
 x35 -0.788   0.167  7
x631 -0.784   0.183  7
x506 -0.778   0.206  6
x475 -0.777   0.211  7
x613 -0.777   0.211  9
x595 -0.766   0.262  9
x338 -0.763   0.277  8
x253 -0.763   0.277 11
x724 -0.762   0.280  6
 x38 -0.762   0.280  8
x402 -0.761   0.286  6
x545 -0.758   0.297  6
x501 -0.758   0.307  5
x921 -0.756   0.318  7
x125 -0.754   0.324  6
x314 -0.749   0.363 12
 x41 -0.746   0.378  7
x803 -0.740   0.419  9
x405 -0.738   0.431  8
      stat p.upper np 
x233 0.865   0.011  9
x544 0.847   0.021  6
x432 0.836   0.032 12
x625 0.828   0.047 10
x429 0.823   0.063 10
x845 0.818   0.076 10
x767 0.818   0.077  9
x241 0.812   0.089  9
x830 0.810   0.096 13
x796 0.810   0.096 12
x256 0.800   0.124  7
 x48 0.784   0.178 11
x165 0.778   0.197 13
x864 0.771   0.229  7
x615 0.763   0.271 11
x860 0.755   0.318 14
x240 0.743   0.404 12
x135 0.732   0.465 10
x280 0.725   0.513 10
x499 0.721   0.529 10
x789 0.719   0.541 11
x376 0.718   0.552 11
x217 0.709   0.607 11
x151 0.703   0.644 11
x268 0.702   0.650 13
$lower:
 [1] "x255" "x193" "x243" "x844" "x339" "x183" "x35"  "x631" "x506" "x475"
[11] "x613" "x595" "x338" "x253" "x724" "x38"  "x402" "x545" "x501" "x921"
[21] "x125" "x314" "x41"  "x803" "x405"

$upper:
 [1] "x233" "x544" "x432" "x625" "x429" "x845" "x767" "x241" "x830" "x796"
[11] "x256" "x48"  "x165" "x864" "x615" "x860" "x240" "x135" "x280" "x499"
[21] "x789" "x376" "x217" "x151" "x268"

[1] 15961 18857 29126
statistics will be computed for all 35 groupings
statistics will be computed for all 5040 permutationss
[1] 0 0
[1] 0 0
statistics will be computed for all 3003 groupings
# comp sep; prop perm with more; ave # per perm; prop perm with any
[1] 3.800000e+01 3.330003e-04 3.546454e-01 2.474192e-01
 obs.num      pval.no avenum.per.perm prop.with.any 
      38 0.0003330003       0.3546454     0.2474192
# comp sep; prop perm with more; ave # per perm; prop perm with any
[1] 38.000  0.000  0.362  0.257
